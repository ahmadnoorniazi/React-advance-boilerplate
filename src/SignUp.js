import './Login.css'
import {connect} from 'react-redux'
import {
  firstName,
  lastName,
  email,
  password,
  onSubmit,
} from './actionCreators/signUp'
import React, {Component} from 'react'

class SignUp extends Component {
  constructor(props) {
    super(props)
  }
  handleSignUpPassword = event => {
    const {dispatch} = this.props
    dispatch(password(event.target.value))
  }
  handleEmail = event => {
    const {dispatch} = this.props

    dispatch(email(event.target.value))
  }
  handleFirstName = event => {
    const {dispatch} = this.props
    dispatch(firstName(event.target.value))
  }
  handleLastName = event => {
    const {dispatch} = this.props
    dispatch(lastName(event.target.value))
  }
  handleSubmit = event => {
    event.preventDefault()
    const {firstName, lastName, email, password, dispatch} = this.props
    if (firstName && lastName && email && password) {
      dispatch(
        onSubmit({
          firstName,
          lastName,
          email,
          password,
        }),
      )
      alert('signUp successfully')
      this.props.history.push('/login')
    }
    alert('please fill out all the fields')
  }
  render() {
    const {firstName, lastName, email, password} = this.props
    return (
      <div className="form-box">
        <div className="head" name="signUpForm">
          SignUp Form
        </div>
        <form onSubmit={this.handleSubmit}>
          <div className="form-group">
            <label className="label-control" htmlFor="firstName">
              First Name
            </label>
            <input
              id="firstName"
              type="text"
              name="firstName"
              placeholder="User Name"
              value={firstName}
              onChange={this.handleFirstName}
            />
          </div>
          <div className="form-group">
            <label className="label-control" htmlFor="lastName">
              Last Name
            </label>
            <input
              id="lastName"
              type="text"
              name="lastName"
              placeholder="User Name"
              value={lastName}
              onChange={this.handleLastName}
            />
          </div>
          <div className="form-group">
            <label className="label-control" htmlFor="email">
              Email
            </label>
            <input
              id="email"
              type="text"
              name="email"
              placeholder="User Name"
              value={email}
              onChange={this.handleEmail}
            />
          </div>
          <div className="form-group">
            <label className="label-control" htmlFor="password">
              Password
            </label>
            <input
              id="password"
              name="password"
              type="password"
              placeholder="password"
              value={password}
              onChange={this.handleSignUpPassword}
            />
          </div>
          <input type="submit" name="Submit" value="Submit" className="btn" />
        </form>
      </div>
    )
  }
}

const mapStateToProps = ({signUp}) => ({
  firstName: signUp.firstName,
  lastName: signUp.lastName,
  password: signUp.password,
  email: signUp.email,
})

export default connect(
  mapStateToProps,
  null,
)(SignUp)
