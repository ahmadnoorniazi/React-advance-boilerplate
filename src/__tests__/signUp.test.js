// these should normally be in your jest setupTestFrameworkScriptFile
import "jest-dom/extend-expect";
import "react-testing-library/cleanup-after-each";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { fireEvent, cleanup, render } from "react-testing-library";
import reducer from "../reducers";
import React from "react";
import SignUp from "../SignUp";

afterEach(cleanup);
test("should test for sign up ", () => {
  const store = createStore(reducer);
  window.alert = function() {};
  const fakeHistory = { push: jest.fn() };
  const { container, getByText, getByLabelText } = render(
    <Provider store={store}>
      <SignUp history={fakeHistory} />
    </Provider>
  );
  const userFirstName = getByLabelText("First Name");
  const userLastName = getByLabelText("Last Name");
  const userEmail = getByLabelText("Email");
  const userPassword = getByLabelText("Password");
  const SubmitNode = getByText("Submit");

  fireEvent.change(userFirstName, { target: { value: "ahmad" } });
  fireEvent.change(userLastName, { target: { value: "noor" } });
  fireEvent.change(userEmail, { target: { value: "ahmad@gmail.com" } });
  fireEvent.change(userPassword, { target: { value: "noor" } });
  fireEvent.submit(SubmitNode);
  expect(SubmitNode.type).toBe("submit");
  expect(fakeHistory.push).toHaveBeenCalledTimes(1);
  expect(fakeHistory.push).toHaveBeenCalledWith("/login");
  expect(container.firstChild).toMatchSnapshot();
});
