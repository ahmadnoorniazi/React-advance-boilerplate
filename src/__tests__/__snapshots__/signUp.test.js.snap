// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test for sign up  1`] = `
<div
  class="form-box"
>
  <div
    class="head"
    name="signUpForm"
  >
    SignUp Form
  </div>
  <form>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="firstName"
      >
        First Name
      </label>
      <input
        id="firstName"
        name="firstName"
        placeholder="User Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="lastName"
      >
        Last Name
      </label>
      <input
        id="lastName"
        name="lastName"
        placeholder="User Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        placeholder="User Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="form-group"
    >
      <label
        class="label-control"
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        placeholder="password"
        type="password"
        value=""
      />
    </div>
    <input
      class="btn"
      name="Submit"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
